src/Confluent.Kafka/Consumer.cs

```diff
@@ -541,7 +541,9 @@ internal Consumer(ConsumerBuilder<TKey, TValue> builder)
             }
 
             var modifiedConfig = config
-                .Where(prop => prop.Key != ConfigPropertyNames.Consumer.ConsumeResultFields);
+                .Where(prop => prop.Key != ConfigPropertyNames.Consumer.ConsumeResultFields)
+                .Concat(Library.NameAndVersionConfig)
```

Comment:
The way I'm doing this in librdkafka and Go is to allow the application (or rather a high-high-level binding) to override the default client.software.*, I think the app/binding config should take precedence in this case.

Reply:
yeah, good point. i wasn't doing this because they're hidden, but it's a good feature. I've reversed the order of the concatenation (changes order `rd_kafka_conf_set` is called) which I believe should achieve this without relying on a librdkafka implementation detail that may change.

feel free to confirm before I merge (will hold off for a bit).